server/src/main/java/org/opensearch/index/translog/transfer/TranslogTransferManager.java

```diff
@@ -106,6 +106,29 @@ public boolean transferSnapshot(TransferSnapshot transferSnapshot, TranslogTrans
                     latchedActionListener
                 )
             );
+
+            // Different listener of metadata as we are not adding it to fileTransferTracker
+            ActionListener<TransferFileSnapshot> mdListener = new ActionListener<>() {
+                @Override
+                public void onResponse(TransferFileSnapshot response) {
+                    latch.countDown();
+                }
+
+                @Override
+                public void onFailure(Exception e) {
+                    logger.info("failed to upload metadata", e);
+                    exceptionList.add(e);
+                    latch.countDown();
+                }
+            };
+
+            transferService.uploadBlobAsync(
```

Comment:
Since we are not waiting for translog upload to succeed before metadata upload, can we end up with metadata files in remote store which are not relevant or needs to be cleaned up? I know during read we are taking care of this but anything we are planning to do for cleanup of these metadata files?

Reply:
That is a great question .  Yes, we are doing that already in `TranslogTransferManager#deleteStaleTranslogMetadataFilesAsync` where we delete all the metadata files except the last referenced one. 

However we do need to take care in the clean up that we are not deleting the last completely uploaded metadata . I would make those changes and reraise the PR .